core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -366,6 +366,17 @@ object SparkSubmit extends CommandLineUtils {
       }.orNull
     }
 
+    // In yarn-cluster mode, download remote '--jars' and
+    // primaryResource in order to get correct hadoop token.
```

Comment:
Can you please change to "In yarn cluster mode, download remote jars and primary resource to local, these jars will be added to classpath of YARN client to get tokens."?